common/fitparser.cpp

```diff
@@ -0,0 +1,1090 @@
+        UString acmInfo;
+        if (header->module_subtype() == intel_acm_t::MODULE_SUBTYPE_TXT) {
+            acmInfo = "TXT ACM ";
+        }
+        else if(header->module_subtype() == intel_acm_t::MODULE_SUBTYPE_STARTUP) {
+            acmInfo = "Startup ACM ";
+        }
+        else if (header->module_subtype() == intel_acm_t::MODULE_SUBTYPE_BOOT_GUARD) {
+            acmInfo = "BootGuard ACM ";
+        }
+        else {
+            acmInfo = usprintf("Unknown ACM (%04Xh)", header->module_subtype());
+            msg(usprintf("%s: Intel ACM with unknown subtype %04Xh found", __FUNCTION__, header->module_subtype()), parent);
+        }
+        
+        acmInfo += usprintf("found at base %Xh\n"
+                            "ModuleType: %04Xh\n"
+                            "ModuleSubtype: %04Xh\n"
+                            "HeaderSize: %08Xh\n"
+                            "HeaderVersion: %08Xh\n"
+                            "ChipsetId: %04Xh\n"
+                            "Flags: %04Xh\n"
+                            "ModuleVendor: %04Xh\n"
+                            "Date: %02X.%02X.%04X\n"
+                            "ModuleSize: %08Xh\n"
+                            "AcmSvn: %04Xh\n"
+                            "SeSvn: %04Xh\n"
+                            "CodeControlFlags: %08Xh\n"
+                            "ErrorEntryPoint: %08Xh\n"
+                            "GdtMax: %08Xh\n"
+                            "GdtBase: %08Xh\n"
+                            "SegmentSel: %08Xh\n"
+                            "EntryPoint: %08Xh\n"
+                            "KeySize: %08Xh\n"
+                            "ScratchSpaceSize: %08Xh\n",
+                            model->base(parent) + localOffset,
+                            header->module_type(),
+                            header->module_subtype(),
+                            header->header_size() * (UINT32)sizeof(UINT32),
+                            header->header_version(),
+                            header->chipset_id(),
+                            header->flags(),
+                            header->module_vendor(),
+                            header->date_day(), header->date_month(), header->date_year(),
+                            header->module_size() * (UINT32)sizeof(UINT32),
+                            header->acm_svn(),
+                            header->se_svn(),
+                            header->code_control_flags(),
+                            header->error_entry_point(),
+                            header->gdt_max(),
+                            header->gdt_base(),
+                            header->segment_sel(),
+                            header->entry_point(),
+                            header->key_size() * (UINT32)sizeof(UINT32),
+                            header->scratch_space_size() * (UINT32)sizeof(UINT32));
+        
+        // Add RsaPublicKey
+        if (header->_is_null_rsa_exponent() == false) {
```

Comment:
!header->_is_null_rsa_exponent()

Reply:
I do not like it, because in `!header` is invisible, as `!` and `h` are both `|`-shaped. 
Will probably swap the order to be `false == ...` to prevent occasional ==/= confusion.